gemd/entity/bounds/integer_bounds.py

```diff
@@ -58,5 +78,10 @@ def contains(self, bounds: Union[BaseBounds, "BaseValue"]) -> bool:
             bounds = bounds._to_bounds()
         if not isinstance(bounds, IntegerBounds):
             return False
+        if (bounds.default_units == DIMENSIONLESS) ^ (self.default_units == DIMENSIONLESS):
+            raise ValueError("Unit mismatch, cannot compare dimensional and dimensionless bounds")
```

Comment:
This is one thing I wanted to discuss and why I marked this PR as a draft.  It seems odd to compare bounds that have dimensions with bounds that are dimensionless.  If this is alright, then I'll remove this check.

Reply:
When we refer to a dimensionless number, it actually has a unit of `1`.  

[IUPAC Green Book](https://iupac.org/wp-content/uploads/2019/05/IUPAC-GB3-2012-2ndPrinting-PDFsearchable.pdf#page=25)

Reply:
It makes perfect sense to compare two dimensionless bounds to each other.  The `^` in the code above is an xor.  If both bounds use defined units, or they both use dimensionless units this will not raise an exception.  I'm thinking about comparing values of defined units with dimensionless units.  Again, this isn't my field so I'm happy to remove this check if it's incorrect.

Reply:
I expressed that badly (and might have been confused when I wrote that).  Sorry.

The `IncompatibleUnitsError` thrown by the convert function will be raised if the units don't align, including if one is dimensionless and the other is not.  So that case is already handled.  For the code you wrote, there's a risk that one bounds has units `''` and one has `'dimensionless'`, both of which refer to dimensionless quantities but would return inconsistent comparisons to the `DIMENSIONLESS` constant.